DigitalLearningSolutions.Data.Tests/DataServices/GroupsDataServiceTests.cs

```diff
@@ -446,6 +446,178 @@ public void DeleteGroup_deletes_group()
+        public void GetGroupCourse_returns_expected_course()
+        {
+            // Given
+            var expectedDateTime = new DateTime(2019, 11, 15, 13, 53, 26, 510);
+            var expectedGroupCourse = GroupTestHelper.GetDefaultGroupCourse(
+                groupCustomisationId: 25,
+                groupId: 103,
+                supervisorAdminId: 1,
+                completeWithinMonths: 0,
+                supervisorFirstName: "Kevin",
+                supervisorLastName: "Whittaker (Developer)",
+                addedToGroup: expectedDateTime);
+
+            // When
+            var result = groupsDataService.GetGroupCourse(25, 103, 101);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.Should().NotBeNull();
+                result.Should().BeEquivalentTo(expectedGroupCourse);
+            }
+        }
+
+        [Test]
+        public async Task DeleteGroupCustomisation_deletes_expected_group_customisation()
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                var groupId = 103;
+                var customisationId = 14675;
+                AddDelegateGroupCustomisationIfNotExist(groupId, customisationId);
+                var groupCustomisationId = (await connection.GetCustomisationsForGroup(groupId))
+                    .OrderByDescending(o => o)
+                    .First();
+
+                // When
+                groupsDataService.DeleteGroupCustomisation(groupCustomisationId);
+
+                // Then
+                var groupCustomisations = await connection.GetCustomisationsForGroup(groupId);
+                groupCustomisations.Should().NotContain(groupCustomisationId);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [TestCase("self_enrolled", 5, 15937, 299238, 1, 0, false, 0, false)]
+        [TestCase("course_started", 5, 25918, 299238, 3, 5, false, 0, false)]
+        [TestCase("course_completed", 5, 25918, 299238, 3, 5, true, 80, true)]
+        [TestCase("non_group_candidate", 5, 25918, 298272, 3, 0, false, 0, false)]
+        public async Task RemoveRelatedProgressRecordsForCourse_does_not_update_for_case(string testCase, int groupId, int customisationId, int candidateId,
+            int enrollmentMethod, int loginCount, bool deleteStartedEnrolment, int progressDuration, bool isCompleted)
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                var progressStarted = new DateTime(2021, 11, 5, 22, 23, 24, 567);
+                var progressRemoved = new DateTime(2021, 11, 6, 22, 23, 24, 567);
+                var progressCompleted = isCompleted ? new DateTime(2021, 11, 7, 22, 23, 24, 567) : (DateTime?)null;

```

Comment:
Can safely move these definitions out of the `try` block (which does inconveniently split up the `// Given` section, but I think that is acceptable (I imagine if the other comments are tackled, then there won't need to be any DB calls in that section))